### PR TITLE
Application commands followup fix

### PR DESCRIPTION
**Pull request by PenguinDevs**
## Summary

This is a follow-up of pull request #634 which was to fix the issue of Pycord bulk upserting global commands every single time the bot was started. This pull request is a follow-up for it to fix the issue of subcommands causing an error in the `register_commands` method.

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

File: discord/bot.py [hunk 1]
```diff
@@ -196,105 +196,161 @@ def get_application_command(
...
+            },
+            # specific to type: 3 - 10 options
+            3: default_props_choice,
+            4: default_props_choice,
+            5: default_props_choice,
+            6: default_props_choice,
+            7: default_props_choice,
+            8: default_props_choice,
+            9: default_props_choice,
+            10: default_props_choice
```
Comment (Dorukyum) [inline discord/bot.py:256]:
```suggestion
            10: default_props_choice,
```

File: discord/bot.py [hunk 2]
```diff
@@ -196,105 +196,161 @@ def get_application_command(
...
+        }
+        missing_props_cmd = {
+            # specific to type: 1 - 3 app command types
+            1: default_props_cmd_type,
+            2: default_props_cmd_type,
+            3: default_props_cmd_type
```
Comment (Dorukyum) [inline discord/bot.py:262]:
```suggestion
            3: default_props_cmd_type,
```

File: discord/bot.py [hunk 3]
```diff
@@ -196,105 +196,161 @@ def get_application_command(
...
+
+        for name in registered_commands_dict.keys():
             if not name in global_pending_application_commands_dict.keys():
                 # When a registered global command is not available in the pending global commands
-                needs_bulk = True
+                needs_update = True 
+
+        await self.register_commands(commands_to_bulk, needs_update)
+
+    async def register_commands(self, commands, needs_update) -> None:
```
Comment (Dorukyum) [inline discord/bot.py:335]:
```suggestion
    async def register_commands(
        self,
        commands: List[ApplicationCommand],
        needs_update: bool = False,
    ) -> None:
```
Maybe also document needs_update

File: discord/bot.py [hunk 4]
```diff
@@ -196,105 +196,161 @@ def get_application_command(
...
+        """|coro|
+        Registers all global commands in :list:`commands` if :bool:`needs_update` is true
+        and guild commands that have been added through :meth:`.add_application_command`.
+
+        This method cleans up all commands over the API and should sync them with the internal cache of commands.
+        .. versionadded:: 2.0
+
+        Parameters
+        -----------
+        commands: :list:
```
Comment (Dorukyum) [inline discord/bot.py:345]:
```suggestion
        commands: List[:class:`.ApplicationCommand`]
```

**Comment by BobDotCom** [comment]
Since I've been working on my own version of application command registration, I'm going to close this PR for now and you can apply the fix to my PR if the issue still exists. I'll provide a link to it here once I open it.

**Comment by BobDotCom** [comment]
Here's my PR: #759

**Pull request closed without merging**